kratos/tests/test_model_part.py

```diff
@@ -803,6 +843,7 @@ def test_create_non_existing_element(self):
     def test_create_non_existing_condition(self):
         current_model = KratosMultiphysics.Model()
         model_part = current_model.CreateModelPart("Main")
+        model_part.CreateNewNode(1, 0.0, 0.0, 0.0)
 
         props = model_part.CreateNewProperties(0)
 

```

Comment:
@philbucher do you have any idea how to compare this error massage? To my understanding it is also dynamic.

Reply:
There are checks for errors: KRATOS_CHECK_EXCEPTION_IS_THROWN

Reply:
Yep but this is python. And the issue is that the error massage is really long and also dependent on which element is registered. Thus, dependent on each application. That's why I can not really compare the string of the error massage.

```
        with self.assertRaisesRegex(RuntimeError, 'Error: The Condition "SomeCertainlyNonExistingCondition" is not registered!\nMaybe you need to import the application where it is defined\?\nThe following Conditions are registered:\n'):
            model_part.CreateNewCondition("SomeCertainlyNonExistingCondition", 1, [1], props)
```

Reply:
you don't need to compare the entire string, thats why the function has `Regex` in its name ;)

Reply:
in any case this should have nothing to do with what you are adding, why do you need to change it?

Quite the opposite I would say that your function should also check in the py-interface if what you want to construct (elem or cond) is in the `KratosComponents`. Otherwise it will segfault if you create sth that doesn't exist

Reply:
I changed it to keep consistency of all the three functions. It is pretty confusing when python and c++ do different things.